src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt

```diff
@@ -92,10 +92,10 @@ constexpr uint16_t kByteSpanSizeLengthInBytes = 2;
     }
 
 
-#define GET_CLUSTER_RESPONSE_CALLBACKS(name)                                                                                               \
+#define GET_CLUSTER_RESPONSE_CALLBACKS(name)                                                                                       \
     Callback::Cancelable * onSuccessCallback = nullptr;                                                                            \
     Callback::Cancelable * onFailureCallback = nullptr;                                                                            \
-    NodeId sourceIdentifier                  = reinterpret_cast<NodeId>(commandObj);                                               \
+    NodeId sourceIdentifier                  = reinterpret_cast<NodeId>(reinterpret_cast<const chip::app::Command *>(commandObj)); \
```

Comment:
Why is the cast to `Command*` a `reinterpret_cast`?  But also, why is that cast needed?  I'd expect that 
```
     reinterpret_cast<NodeId>(reinterpret_cast<const chip::app::Command *>(commandObj)) == reinterpret_cast<NodeId>(commandObj)
```
tests true, since all the casts are reinterpret_cast....  So either that inner cast is not needed at all or it should be a different kind of cast (and we should document why it's needed).

Reply:
Previously I think the reinterpret_cast acts like static_cast for casting between base classes. Seems I'm wrong.
I realized that instead of changing this, I should change the code in controller/ChipDevice to use CommandSender.

Anyway, I will change this part to match ReadClient later.